packages/type-definitions/src/subtype.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type { RegistryTypes } from '@polkadot/types/types'
+
+/**
+ * Removes parts of the type definition without altering the original object.
+ *
+ * @param types The types that should be changed.
+ * @param keys The keys that should get removed.
+ * @returns The adjusted types.
+ */
+export function subtype(types: RegistryTypes, keys: string[]): RegistryTypes {
+  const adjustedTypes = { ...types }
+  keys.forEach((key) => {
+    delete adjustedTypes[key]
+  })
+  return adjustedTypes
+}
```

Comment:
we could also make it accept additional types as a parameter so it's 
```js
x = subtype(y, {newType: 'def'}, ['removeMe'])
```
instead of 
```js
x = {...subtype(y, ['removeMe']), newType: 'def'}
```
What do you think?

Reply:
Good idea. Did that as well! [b1f8a41](https://github.com/KILTprotocol/sdk-js/pull/687/commits/b1f8a41bd410da38930de2eb5a2178c0c5065075)